.gitlab-ci.yml

```diff
@@ -0,0 +1,117 @@
+variables:
+  DEFAULT_BRANCH: 'main'
+  GITHUB_REPO_URL:
+    description: "The Github Repo URL for release-please, in the format of 'owner/repo'"
+    value: 'NorthernTechHQ/nt-gui'
+  GITHUB_USER:
+    description: 'The Github user for release-please'
+    value: 'mender-test-bot'
+  GITHUB_USER_NAME:
+    description: 'The Github username for release-please'
+    value: 'mender-test-bot'
+  GITHUB_USER_EMAIL:
+    description: 'The Github user email for release-please'
+    value: 'mender@northern.tech'
+  RUN_RELEASE:
+    description: 'Run a new release'
+    value: 'false'
+    options:
+      - 'true'
+      - 'false'
+  GIT_CLIFF:
+    description: 'Run git cliff to override the release-please changelog'
+    value: 'true'
+    options:
+      - 'true'
+      - 'false'
+  NODE_IMAGE:
+    value: 'node:22-alpine'
+    description: 'Node version to use for building and testing'
+
+stages:
+  - test
+  - changelog
+  - release
+
+include:
+  - project: 'Northern.tech/Mender/mendertesting'
+    file:
+      - '.gitlab-ci-github-status-updates.yml'
+
+lint:
+  stage: test
+  image: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/${NODE_IMAGE}
+  needs: []
+  script:
+    - npm ci
+    - npm run lint
+  tags:
+    - hetzner-amd-beefy
+
+changelog:
+  image: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/node:20
+  stage: changelog
+  tags:
+    - hetzner-amd-beefy
+  rules:
+    - if: $RUN_RELEASE == "true"
+      when: never
+    - if: $CI_COMMIT_BRANCH =~ "/^\d+\.\d+\.x$/"
+    - if: $CI_COMMIT_BRANCH == $DEFAULT_BRANCH
+  before_script:
+    # install release-please
+    - npm install -g release-please
+    # install github-cli
+    - |
+      mkdir -p -m 755 /etc/apt/keyrings \
+        && wget -qO- https://cli.github.com/packages/githubcli-archive-keyring.gpg | tee /etc/apt/keyrings/githubcli-archive-keyring.gpg > /dev/null \
+        && chmod go+r /etc/apt/keyrings/githubcli-archive-keyring.gpg \
+        && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | tee /etc/apt/sources.list.d/github-cli.list > /dev/null \
+        && apt update \
+        && apt install gh jq -y
+    # Setting up git
+    - |
+      echo "INFO - setting up git"
+      git config --global user.email "${GITHUB_USER_EMAIL}"
+      git config --global user.name "${GITHUB_USER_NAME}"
+    - npm install -g git-cliff
+    # GITHUB_TOKEN for Github cli authentication
+    - export GITHUB_TOKEN=${GITHUB_CLI_TOKEN}
+    # getting the centralized git cliff config
+    - wget --output-document cliff.toml https://raw.githubusercontent.com/mendersoftware/mendertesting/master/utils/cliff.toml
```

Comment:
Another note: here you maybe require the `cliff.toml.scoped` file, because you're having multiple services to scope out in this repo